azure/services/subnets/subnets.go

```diff
@@ -104,23 +112,29 @@ func (s *Service) Delete(ctx context.Context) error {
 
 	if !s.Scope.IsVnetManaged() {
 		log.V(4).Info("Skipping subnets deletion in custom vnet mode")
-
-		s.Scope.UpdateDeleteStatus(infrav1.SubnetsReadyCondition, serviceName, nil)
 		return nil
 	}
 
-	var result error
+	specs := s.Scope.SubnetSpecs()
+	if len(specs) == 0 {
+		return nil
+	}
 
 	// We go through the list of SubnetSpecs to delete each one, independently of the result of the previous one.
 	// If multiple errors occur, we return the most pressing one.
 	//  Order of precedence (highest -> lowest) is: error that is not an operationNotDoneError (i.e. error deleting) -> operationNotDoneError (i.e. deleting in progress) -> no error (i.e. deleted)
-	for _, subnetSpec := range s.Scope.SubnetSpecs() {
+	var result error
+	for _, subnetSpec := range specs {
 		if err := s.DeleteResource(ctx, subnetSpec, serviceName); err != nil {
 			if !azure.IsOperationNotDoneError(err) || result == nil {
 				result = err
 			}
 		}
 	}
-	s.Scope.UpdateDeleteStatus(infrav1.SubnetsReadyCondition, serviceName, result)
+
+	if s.Scope.IsVnetManaged() {
```

Comment:
we don't actually need this check since we're already returning early on line 115 when the vnet isn't managed